internal/distributed/datanode/client/client_test.go

```diff
@@ -125,3 +130,89 @@ func Test_NewClient(t *testing.T) {
 	err = client.Close()
 	assert.NoError(t, err)
 }
+
+func Test_InternalTLS(t *testing.T) {
+	paramtable.Init()
+	validPath := "../../../../configs/cert1/ca.pem"
+	ctx := context.Background()
+	client, err := NewClient(ctx, "test", 1)
```

Comment:
Also, here I think I have written the test completely wrong.

I should test the NewClient() with TLS configs before replacing it with mock client.
https://github.com/milvus-io/milvus/blob/4d08eec1af6a9ca38d29f642000cc4a5113ec488/internal/distributed/proxy/service_test.go#L959

This will successfully create the NewClient().

How should I test if the NewClient is behaving in the same manner it is supposed to work?
1 option I can think is of writing a mock datanode server and try to connect to that.Are there any better suggestions in your mind @xiaofan-luan

Reply:
one thing you can do is to add an integration test in tests/integration.

You can create a MiniCluster and try to connect between

Reply:
I went through the code in tests/integration. I am seeing a mini cluster already present.

I'll create a separate test suite for internal TLS, including necessary certificates. This suite will cover basic operations like creating collections, inserting data, and running queries to ensure secure communication between nodes.This way we can validate TLS functionality without affecting other test cases